### PR TITLE
Added Profile view

### LINKED ISSUE #5: User should be able to view profiles
As a registered user, with an account linked; I should be able to run `/profile` to view an embed with my profile details in it.

### Scenario 0: Own profile (Registered)
1. As a registered user I type `/profile`
2. The bot replies with an embed of my profile

### Scenario 1: Own profile (Not registered)
1. As a1 non-registered user I type `/profile`
2. The bot replies informing me my profile is not linked

### Scenario 2: Someone else's profile (Registered)
1. As a user I type `/profile @zkldi#2965`
2. The bot replies with an embed of their profile

### Scenario 3: Someone else's profile (Not registered)
1. As a user I type `/profile @zkldi#2965`
2. The bot replies informing me that that user has not linked their profile

### PR DESCRIPTION
**Pull request by Puffycheeses**
Fetch a users details and scores and provide a "ui" for viewing specific game details.
| | |
|-|-|
|image|image|
| image|image|

There are a few things that still need work:
- Relating discord user `@` to a kamai user ID
- Data validation, Ensure we actually got users info

Closes #5

### CODE REVIEW

**Comment by Puffycheeses** [comment]
PR is in a position where It's ready for review

File: .eslintrc.json [hunk 1]
```diff
@@ -18,7 +18,10 @@
     "rules": {
         "indent": [
             "error",
-            "tab"
+            "tab",
+            {
+                "SwitchCase":  1
```
Comment (Puffycheeses) [inline .eslintrc.json:23]: Eslint doesn't want Switches to be indented. This fixes that :+1:

File: src/utils/utils.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+};
+
+export const getPfpUrl = (userId: number): string => {
+	if (ProcessEnv.ENV !== "prod") {
+		return `https://kamaitachi.xyz/static/images/users/${userId}-pfp.png`;
+	}
+	return PrependTachiUrl(`/users/${userId}/pfp`);
+};
+
+/** @TODO @zkldi is there somewhere this exists already? */
```
Comment (Puffycheeses) [inline src/utils/utils.ts:47]: @zkldi

File: src/utils/utils.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+export const formatGameWrapper = (game: SimpleGameType<Game>): string => FormatGame(game.game, game.playtype);
+
+export const prettyRatingString = <I extends IDStrings = IDStrings>(rating: UGSRatingsLookup[I]): string => {
+	switch (rating) {
+		case "ktRating":
+			return "Kamai Rating";
+		case "ktLampRating":
+			return "Kamai Lamp Rating";
+		case "sieglinde":
+			return "???";
```
Comment (Puffycheeses) [inline src/utils/utils.ts:28]: @zkldi Need input on this value!

Comment (Puffycheeses) [reply src/utils/utils.ts:28]: Actually input on all these values would be appreciated.

**Review by Puffycheeses — COMMENTED** [review]
Self review good, Added a few comments for other reviewers and updated where I spotted issues.

**Review by zkldi — CHANGES_REQUESTED** [review]

File: src/utils/utils.ts (hunk 2 above)
Comment (Puffycheeses) [reply src/utils/utils.ts:47]: It seemed like online people were saying make an array then gen the type off of the array. Unsure how much we'd use it though. Up to you.

File: src/utils/utils.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+		case "MFCP":
+			return "MFCs";
+		default:
+			return "Skill";
+	}
+};
+
+export const getPfpUrl = (userId: number): string => {
+	if (ProcessEnv.ENV !== "prod") {
+		return `https://kamaitachi.xyz/static/images/users/${userId}-pfp.png`;
```
[the remark this answers is not in the document]
Comment (Puffycheeses) [reply src/utils/utils.ts:42]: Yeah I dont think spinning up the CDN locally is a great idea 😂 but yeah this should work _for now_

I'll add a TODO

[the remark this answers is not in the document]
Comment (zkldi) [reply src/utils/utils.ts:42]: There's an option in `tachi-server`'s conf.json5 to spin the cdn up locally as part of the express server, it should be good for dev.

File: src/utils/utils.ts [hunk 5]
```diff
@@ -50,3 +51,17 @@ export const gamesToChoicesObject = (): [name: string, value: string][] => {
...
 };
+
+export const formatGameScoreRating = <I extends IDStrings = IDStrings>(
+	gpt: SimpleGameType<Game>,
+	ratingName: UGSRatingsLookup[I],
+	value: number
+): string => {
+	const gptConfig = GetGamePTConfig(gpt.game, gpt.playtype);
+	/** @TODO UGSRatingsLookup should be able to cast to ScoreCalculatedDataLookup straight up */
+	const val = gptConfig.scoreRatingAlgFormatters[ratingName as unknown as ScoreCalculatedDataLookup[I]];
```
Comment (zkldi) [inline src/utils/utils.ts:62]: As mentioned on discord: should be profileRatingAlgFormatters (and then no cast is necessary)

**Review by zkldi — CHANGES_REQUESTED** [review]
Just this one change and then I think we're all good!

**Comment by Puffycheeses** [comment]
Should be good to go now :+1:

**Comment by zkldi** [comment]
:+1:

**Pull request merged**